### PR TITLE
Describe core, extended, custom support levels

### PR DESCRIPTION
**Pull request by bowei**
Fixes https://github.com/kubernetes-sigs/service-apis/issues/195

### CODE REVIEW

**Comment by bowei** [comment]
@jpeach @danehans @robscott

File: docs-src/concepts.md [hunk 1]
```diff
@@ -173,7 +173,31 @@ handling this:
 
 * TODO
 
-### Extensibility
+### Conformance
+
+As this API aims to cover a wide set of implementations and use cases,
+it will not be possible for all implementations to support all
+features at a given time. However, we do expect that the set of
```
Comment (jpeach) [inline docs-src/concepts.md:180]: Probably should be:
```
However, we do expect the set of features that are supported to be converging
```

File: docs-src/concepts.md [hunk 2]
```diff
@@ -173,7 +173,31 @@ handling this:
...
+features that are supported to be converging. To model this in the
+API, we are taking a similar approach as with
+[SIG-CONFORMANCE][sig-conformance] work on conformance
```
Comment (jpeach) [inline docs-src/concepts.md:183]: Should `sig-conformance` be a link to somewhere?

File: docs-src/concepts.md [hunk 3]
```diff
@@ -173,7 +173,31 @@ handling this:
...
+profiles. Features as described in the API spec will be divided into
+three major categories:
+
+* **CORE** features are those that will always work in a portable way
+  for ALL implementations that support the API.
```
Comment (hbagdi) [inline docs-src/concepts.md:188]: I was using this as a guiding principle to decide which bucket do the features go into and ran into two problems:

1. If we decide a feature is supported in core, and there comes an implementer who says their proxy can't support feature x, how is that resolved? We need to document this for future-selves or other future maintainers of this API.
2. How is "ALL" defined? Quite a few implementers are working together in this working group but we probably do have some that are not represented.

File: docs-src/concepts.md (hunk 1 above)
Comment (bowei) [reply docs-src/concepts.md:180]: done

File: docs-src/concepts.md (hunk 2 above)
Comment (bowei) [reply docs-src/concepts.md:183]: yes -- this is a broken link

File: docs-src/concepts.md [hunk 4]
```diff
@@ -173,7 +173,31 @@ handling this:
 
 * TODO
 
-### Extensibility
+### Conformance
+
```
[the remark this answers is not in the document]
Comment (bowei) [reply docs-src/concepts.md:177]: Making the language more explicit

File: docs-src/concepts.md (hunk 3 above)
Comment (bowei) [reply docs-src/concepts.md]: Adding some language to address your questions -- please take a look

File: docs-src/concepts.md [hunk 5]
```diff
@@ -173,7 +173,53 @@ handling this:
...
+[sig-arch][sig-arch-bdd] work on conformance profiles. Features as
+described in the API spec will be divided into three major categories:
+
+[sig-arch-bdd]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/960-conformance-behaviors
+
+* **CORE** features will be portable and we expect that there is a
+  reasonable roadmap for ALL implementations towards support of APIs
+  in this category.
+* **EXTENDED** features are those that are portable but not
+  universally supported across implementations.. Those implementations
```
Comment (hbagdi) [inline docs-src/concepts.md:195]: typo: implementations..

File: docs-src/concepts.md (hunk 3 above)
Comment (hbagdi) [reply docs-src/concepts.md]: It resolves some but not more about the process of how to define "ALL", which is a larger topic and seems out of out of scope from this PR.

**Comment by bowei** [comment]
Updated

I am going to send another PR covering what we think the "all" for core will look like.

**Comment by hbagdi** [comment]
Oops. I didn't know it would do that. Maybe I should have just written lgtm.
In hindsight, that makes sense because Bowei is an approver.

**Pull request merged**